Landing page and platform merge

aapeliv opened the pull request:
First cut at merging landing site and app.

TODO:
* ~~Navbar for signed out users~~
* ~~Allowing signed out users to view markdown pages~~
* ~~SEO tags (meta tags, etc)~~
* ~~Main home page~~
* ~~Redirects~~
* ~~Team page~~
* ~~Missing "scroll to learn more"~~

**Web frontend checklist**
- [ ] Formatted my code with `yarn format && yarn lint --fix`
- [ ] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

aapeliv committed:
Move markdown files to app/web from site

aapeliv committed:
First cut at markdown rendering on app

aapeliv committed:
Fix stuff

aapeliv committed:
Move actual page to MarkdownPage and add some frontmatter bits and formatting

aapeliv committed:
Work on styling

aapeliv committed:
Add @types/markdown-it

aapeliv committed:
Fix up essay tags

aapeliv commented:
Old:
<img width="1680" alt="Screen Shot 2022-01-15 at 3 16 26 pm" src="https://user-images.githubusercontent.com/2975660/149623105-1f2fd826-283f-49b0-bdd9-7db3652f3a40.png">


New:
<img width="1680" alt="Screen Shot 2022-01-15 at 3 16 32 pm" src="https://user-images.githubusercontent.com/2975660/149623107-797e48a3-28f4-4dce-b945-7197fc01f1da.png">


Old:
<img width="1680" alt="Screen Shot 2022-01-15 at 3 16 48 pm" src="https://user-images.githubusercontent.com/2975660/149623115-f9bcc76d-6233-47c5-8b94-74eaccb44710.png">

New:
<img width="1680" alt="Screen Shot 2022-01-15 at 3 16 52 pm" src="https://user-images.githubusercontent.com/2975660/149623117-33e294d0-f402-4f50-a949-b1cb8dd8283a.png">


Old:
<img width="1680" alt="Screen Shot 2022-01-15 at 3 18 39 pm" src="https://user-images.githubusercontent.com/2975660/149623304-c6cb98bb-cdc2-4dcc-823d-0e3648bbcb6d.png">

New:
<img width="1674" alt="Screen Shot 2022-01-15 at 3 24 23 pm" src="https://user-images.githubusercontent.com/2975660/149623306-64861a62-29c7-4942-8258-35fdbe8b9664.png">

aapeliv committed:
Fix up soft links

aapeliv committed:
Move LoggedInMenu into its own component

This way we can soon use Navigation.tsx without useCurrentUser that kicks you out if you aren't logged in

aapeliv committed:
Don't display LoggedInMenu if not logged in

aapeliv committed:
Add buttons to sign up and log in when logged out

aapeliv committed:
Implement logged out nav menus

aapeliv commented:
Logged out menus:
<img width="428" alt="Screen Shot 2022-01-15 at 4 41 20 pm" src="https://user-images.githubusercontent.com/2975660/149625817-d07e96b1-5f25-43d6-b458-5c8eba3c434b.png">
<img width="427" alt="Screen Shot 2022-01-15 at 4 41 15 pm" src="https://user-images.githubusercontent.com/2975660/149625824-96fed7c1-1972-4c16-a78b-511fe7fcb90b.png">
<img width="1677" alt="Screen Shot 2022-01-15 at 4 41 02 pm" src="https://user-images.githubusercontent.com/2975660/149625825-30584f76-ddd4-4928-b195-1f622ccb23f6.png">

papeldeorigami committed:
feat: Html Meta for markdown pages

- Description and share image set in html meta tags
- Set default share image

aapeliv commented:
Team page:

old:
<img width="1678" alt="Screen Shot 2022-01-15 at 5 56 09 pm" src="https://user-images.githubusercontent.com/2975660/149628493-235e81fb-4c6a-46f6-9beb-1664c93b79f2.png">

new:
<img width="1680" alt="Screen Shot 2022-01-15 at 5 56 16 pm" src="https://user-images.githubusercontent.com/2975660/149628495-ef1a0f90-6241-453f-96e9-ab8dad981433.png">

aapeliv committed:
Implement team page in react

aapeliv committed:
Clean up

aapeliv committed:
Fix case sensitivity issue

aapeliv commented:
@lucaslcode: I can't get the meta tags to work properly. They're all just the same "Couchers.org Beta" in the vercel preview and when doing `next build && next start` locally.

They are being updated in the DOM, but not in the generated HTML that's sent back from the server.

aapeliv commented:
OK, these pages are straight up just not being SSRd, I don't know what I'm doing wrong or if I'm just very confused here!

lucaslcode reviewed (commented):
Great work! I will look properly later. Your `getStaticPaths` needs to be async and await the slugs function.

lucaslcode reviewed (commented):
Some general stuff:
- There's a bug in AppRoutes so these pages are statically rendering a loader instead of the page. I think line 79 `!isMounted || (isPrivate && !isAuthenticated)` should be `(!isMounted && isPrivate) || (isPrivate && !isAuthenticated)`
- Are you using markdown-it instead of our markdown component because ours doesn't work with ssr? Maybe we should replace that one with markdown-it.
- The tag colours look weird, but I can't offer any suggestions (except for maybe removing them completely?)

aapeliv committed:
Remove soft links

aapeliv committed:
Move contributor docs into web folder

aapeliv commented:
> There's a bug in AppRoutes so these pages are statically rendering a loader instead of the page. I think line 79 `!isMounted || (isPrivate && !isAuthenticated)` should be `(!isMounted && isPrivate) || (isPrivate && !isAuthenticated)`

It works well (on these pages) and is server side rendered (turning off JS) if I just replace that logic with `false` (aka never show the spinner). It seems to be something to do with `isMounted`, I don't know what that does.

aapeliv commented:
>     * Are you using markdown-it instead of our markdown component because ours doesn't work with ssr? Maybe we should replace that one with markdown-it.

I'm just using it because it was used on Vue and I wanted everything to be as similar as possible.

>     * The tag colours look weird, but I can't offer any suggestions (except for maybe removing them completely?)

I think we can leave it for now. The tag colors match the boxes on the current landing page when you scroll down.

aapeliv committed:
Tweak auth pages and add navbar

aapeliv committed:
Work on landing page

aapeliv committed:
Lint fix

aapeliv committed:
Fix up landing page variant issue

aapeliv commented:
@lucaslcode: I've implemented the landing page now. It works as follows:

1. if you're logged in, `/` shows you the dashboard
2. if you're logged out, `/` shows the landing page as currently on couchers.org

I'm out of juice on this at the moment. I know it's a pretty crap PR in terms of code quality...

Random stuff:
* ~~I don't know why Vercel is failing, for all I can see it should work~~
* It's missing a little "scroll down for more" arrow on the landing page
* I tweaked some of the login/signup CSS. It seems it's quite a bit of a mess at the moment (and I doubt I made it any better).
* The landing page AppRoute thing is a mess: if logged in it should be `variant = "standard"`, but if logged out, it should be `full-screen`. This is the `currentVariant` mess.
* This does not follow the UI/UX designs for the landing page. I don't think they're quite ready and thought out yet, and I don't have the skill/patience/juice to build that. So I ported over our existing landing page and made-do. This is (in my opinion) a small regression visually and in UI/UX, but I think it's worth it and we can tweak it further later. I just want to get rid of the awful vue landing site.
* SSR still doesn't work, it's something to do with `isMounted`, if I just comment that out it works fine (but flashes the wrong type of page at times)
* I changed the navbar to be `fixed` and added padding. This gives us pages that are the right height (e.g. login/signup) with just `100vh`, and allows the landing page above-fold to just be `100vh` and then have a nice page below.
* The landing page isn't set up fully with translations yet, I ran out of juice. But it's not a huge thing since I added the `landing` translation component. The translation keys are also a tad bit of a mess in like duplicates between `auth`, etc.
* I bastardized `BasicForm` to put it on the landing page. After you fill it in it'll redirect to `/signup` (but it flashes first). If you've already filled it in it and you goon the landing page again it shows a button to finish signing up.

aapeliv committed:
Fix landing translations

lucaslcode commented on app/web/components/AppRoute.tsx:
```diff
@@ -43,7 +44,7 @@ export const useAppRouteStyles = makeStyles((theme) => ({
   isPrivate: boolean;
   noFooter?: boolean;
-  variant?: "standard" | "full-screen" | "full-width";
+  variant?: "standard" | "full-screen" | "full-width" | "landing";
```
Check

lucaslcode commented on app/web/components/IconText.tsx:
```diff
@@ -19,7 +19,7 @@ const useStyles = makeStyles((theme) => ({
 interface IconTextProps {
   icon: OverridableComponent<SvgIconTypeMap<unknown, "svg">>;
-  text: string;
+  text: ReactNode;
```
Don't include textbody for type=string

lucaslcode commented on app/web/components/Navigation/LoggedInMenu.tsx:
```diff
@@ -0,0 +1,84 @@
+        id="navigation-menu"
+        open={menuOpen}
+        anchorEl={menuRef.current}
+        onClose={() => setMenuOpen(() => false)}
```
(false)

lucaslcode commented on app/web/components/Navigation/Navigation.tsx:
```diff
@@ -128,8 +131,62 @@ const navMenu = (data: MenuGenData): Array<MenuItemProps> => [
+  },
+];
+
+const loggedOutDrawerMenu = (data: MenuGenData): Array<MenuItemProps> => [
```
rename data

lucaslcode commented on app/web/components/Navigation/Navigation.tsx:
```diff
@@ -378,7 +432,7 @@ export default function Navigation() {
   return (
     <AppBar
-      position="sticky"
+      position="fixed"
```
check margins

lucaslcode commented on app/web/features/Index.tsx:
```diff
@@ -0,0 +1,14 @@
+  const { authState } = useAuthContext();
+  const isAuthenticated = authState.authenticated;
+
+  if (isAuthenticated) {
```
Change to a rewrite

lucaslcode commented on app/web/features/auth/locales/en.json:
```diff
@@ -156,5 +156,7 @@
         "title": "Notification settings"
-    }
+    },
+    "sign_in": "Sign in",
+    "create_account": "Create an account"
```
rename? global?

lucaslcode commented on app/web/components/AppRoute.tsx:
```diff
@@ -19,6 +19,7 @@ export const useAppRouteStyles = makeStyles((theme) => ({
   },
   nonFullScreenStyles: {
     height: "100%",
+    padding: "64px 0 0 0",
```
Make navbar effect clearer/use variable

lucaslcode commented on app/web/features/auth/useAuthStyles.ts:
```diff
@@ -94,31 +98,30 @@ const useAuthStyles = makeStyles((theme) => ({
     flexDirection: "column",
     display: "flex",
-    justifyContent: "flex-end",
+    // justifyContent: "flex-end",
```
rm

lucaslcode commented on app/web/features/landing/LandingPage.tsx:
```diff
@@ -0,0 +1,521 @@
+  topSection: {
+    padding: theme.spacing(3),
+  },
+  section: {
+    marginLeft: "auto",
+    marginRight: "auto",
+    maxWidth: "769px",
+    margin: theme.spacing(3),
+    marginTop: theme.spacing(12),
+    [theme.breakpoints.down("md")]: {
+      marginTop: theme.spacing(6),
+    },
+  },
```
MUI?

lucaslcode commented on app/web/features/markdown/MarkdownPage.tsx:
```diff
@@ -0,0 +1,182 @@
+import HtmlMeta from "components/HtmlMeta";
+import markdown from "markdown-it";
+
+const mkd = new markdown();
```
Make issue to replace other md renderer or do here if easy

lucaslcode commented on app/web/features/markdown/MarkdownPage.tsx:
```diff
@@ -0,0 +1,182 @@
+  root: {
+    marginLeft: "auto",
+    marginRight: "auto",
+    maxWidth: "769px",
```
container mui

lucaslcode commented on app/web/features/team/Team.tsx:
```diff
@@ -0,0 +1,139 @@
+  tightSection: {
+    marginLeft: "auto",
+    marginRight: "auto",
+    maxWidth: "769px",
```
mui

lucaslcode commented on app/web/features/team/Team.tsx:
```diff
@@ -0,0 +1,139 @@
+    flex: "1 0 auto",
+  },
+  avatar: {
+    width: "96px",
+    height: "96px",
```
no magic

lucaslcode commented on app/web/features/team/team.json:
```diff
@@ -0,0 +1,284 @@
+  },
+  {
+    "name": "Lucas Levin",
+    "role": "Head of Frontend (web)",
+    "location": "London, United Kingdom",
```
change

lucaslcode reviewed (commented):
This is notes for myself

lucaslcode committed:
test rewrites

lucaslcode committed:
switch to middleware, some style fixes

lucaslcode committed:
tweak landing page styles

lucaslcode committed:
fix some styling

lucaslcode committed:
Merge remote-tracking branch 'origin/develop' into web/feature/landing-page-merge

lucaslcode committed:
add redirects

lucaslcode commented:
This is basically done but needs some checking

lucaslcode commented:
Note that the landing page/dashbiard redirect middleware doesn't work on preview, because the cookie domain is .couchershq.org, so web browsers don't include it in the request.

lucaslcode committed:
add markdown-it types

lucaslcode committed:
add missing types

lucaslcode committed:
update footer links to relative

lucaslcode committed:
add read more button

lucaslcode committed:
clean up MarkdownPage

lucaslcode committed:
clean up namespaces

lucaslcode reviewed (approved):
This should be good once all checks are green

lucaslcode reviewed (changes requested):
Wait the home page still isn't pre-rendered

lucaslcode committed:
don't display loader on non-private pages

lucaslcode reviewed (approved):
Okay NOW when checks are green
